docs/docs/adding-search-with-algolia.md

```diff
@@ -2,593 +2,579 @@
-
-```text:title=.env.example
-# rename this file to .env and supply the values listed below
-# also make sure they are available to the build tool (e.g. Netlify)
-# warning: variables prefixed with GATSBY_ will be made available to client-side code
-# be careful not to expose sensitive data (in this case your Algolia admin key)
-
-GATSBY_ALGOLIA_APP_ID=insertValue
-GATSBY_ALGOLIA_SEARCH_KEY=insertValue
-ALGOLIA_ADMIN_KEY=insertValue
-```
-
-The `queries` allow you to grab the data you want Algolia to index directly from Gatsby's GraphQL layer by exporting from `src/utils/algolia.js` an array of objects, each containing a required GraphQL query and an optional index name, transformer function and settings object.
-
-> Note: The excerpts below are pruned to 5000 characters. This is because, by default, Algolia has an upper bound of 10KB for an index entry. If you hit this boundary, you'll be presented with an error like this: `AlgoliaSearchError: Record at the position XX objectID=xx-xx-xx-xx-xx is too big size=10356 bytes. Contact us if you need an extended quota`. Make sure that your overall record is within this limit.
-
-```js:title=src/utils/algolia.js
+```js:title=src/utils/algolia-queries.js
 const pageQuery = `{
   pages: allMarkdownRemark(
     filter: {
-      fileAbsolutePath: { regex: "/pages/" },
-      frontmatter: {purpose: {eq: "page"}}
+      fileAbsolutePath: { regex: "/${pagePath.replace("/", "/")}/" },
     }
   ) {
     edges {
       node {
-        objectID: id
+        id
         frontmatter {
           title
-          slug
         }
-        excerpt(pruneLength: 5000)
-      }
-    }
-  }
-}`
-
-const postQuery = `{
-  posts: allMarkdownRemark(
-    filter: { fileAbsolutePath: { regex: "/posts/" } }
-  ) {
-    edges {
-      node {
-        objectID: id
-        frontmatter {
-          title
+        fields {
           slug
-          date(formatString: "MMM D, YYYY")
-          tags
         }
         excerpt(pruneLength: 5000)
       }
     }
   }
 }`
 
-const flatten = arr =>
-  arr.map(({ node: { frontmatter, ...rest } }) => ({
+function pageToAlgoliaRecord({ node: { id, frontmatter, fields, ...rest } }) {
+  return {
+    objectID: id,
     ...frontmatter,
+    ...fields,
     ...rest,
-  }))
-const settings = { attributesToSnippet: [`excerpt:20`] }
+  }
+}
 
 const queries = [
   {
     query: pageQuery,
-    transformer: ({ data }) => flatten(data.pages.edges),
-    indexName: `Pages`,
-    settings,
-  },
-  {
-    query: postQuery,
-    transformer: ({ data }) => flatten(data.posts.edges),
-    indexName: `Posts`,
-    settings,
+    transformer: ({ data }) => data.pages.edges.map(pageToAlgoliaRecord),
+    indexName,
```

Comment:
This throws an error for me: `ReferenceError: indexName is not defined`

Reply:
Part of the same two lost lines as `pagePath`; should be `const indexName = "Pages"`